test/benchmarks/material/button-harness/testbed-benchmark-utilities.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Records the performance of the given function.
+ *
+ * @param id A unique identifier.
+ * @param callback A function whose performance will be recorded.
+ */
+export async function benchmark(id: string, callback: Function) {
+  const t0 = performance.now();
+  await callback();
+  const t1 = performance.now();
+  console.warn(`${id}: ${t1 - t0}`);
+}
+
+export function getButtonWithText(text: string): HTMLButtonElement {
+  const xpathExpression = `//button[//span[text()=' ${text} ']]`;
+  const xPathResult = document.evaluate(
```

Comment:
We should be able to avoid this XPath code here. Seems pretty uncommon to me. Protractor uses `document.querySelector` too. See: https://github.com/angular/protractor/blob/master/lib/clientsidescripts.js#L627

So from a performance comparison side, it should be better being equivalent here. Plus it would be simpler